src/commands/CodingJamsCommand.ts

```diff
@@ -0,0 +1,307 @@
+	}
+
+	startScheduler(client: Client) {
+		jamSchedulerTick(client);
+		// every 30 min
+		cron.schedule("*/30 * * * *", jamSchedulerTick.bind(this, client));
+	}
+
+	async execute(interaction: ChatInputCommandInteraction): Promise<void> {
+		if (!(await hasAdminPerms(interaction))) {
+			await interaction.reply({
+				content: "The coding jam commands are meant to be used by admins only.",
+				ephemeral: true,
+			});
+			return;
+		}
+
+		const name = interaction.options.getString("name") || "Default Name";
+		const proposal = interaction.options.getString("proposal") || "";
+
+		const numVotes = interaction.options.getInteger("num-votes") || 3;
+		const numProposals = interaction.options.getInteger("num-proposals") || 3;
+		const selectionType =
+			interaction.options.getString("proposal-selection-type") || pollSelectionTypes.topAll;
+
+		const subCmdGroup = interaction.options.getSubcommandGroup() || "cmd";
+		const subCmd = interaction.options.getSubcommand();
+
+		let startDate: DateTime | null;
+		let endDate: DateTime | null;
+		let duration: Duration | null;
+
+		if (subCmd === "new") {
+			startDate = await checkDate(interaction, interaction.options.getString("start-date"));
+			endDate = await checkDate(interaction, interaction.options.getString("end-date"));
+			duration = await checkDuration(interaction, interaction.options.getString("duration"));
+
+			if (!startDate) return;
+
+			if (!(endDate || duration) && subCmdGroup === "poll") {
+				await interaction.reply({
+					content: "You must provide either an end date or a duration...",
+					ephemeral: true,
+				});
+				return;
+			}
+
+			if (subCmdGroup === "poll") endDate = endDate || startDate.plus(duration!);
+		} else if (subCmd === "extend") {
+			endDate = await checkDate(interaction, interaction.options.getString("end-date"));
+
+			if (!endDate) return;
+		}
+
+		if (subCmdGroup === "jam") {
+			switch (subCmd) {
+				case "new":
+					newJam(interaction, name, proposal, startDate!);
+					break;
+				case "extend":
+					editJam(interaction, name, endDate!);
+					break;
+				case "view":
+					viewJam(interaction, name);
+					break;
+				case "delete":
+					deleteJam(interaction, name);
+					break;
+			}
+		} else if (subCmdGroup === "poll") {
+			switch (subCmd) {
+				case "new":
+					newPoll(interaction, name, numProposals, numVotes, selectionType, startDate!, endDate!);
+					break;
+				case "extend":
+					editPoll(interaction, name, endDate!);
+					break;
+				case "view":
+					viewPoll(interaction, name);
+					break;
+				case "delete":
+					deletePoll(interaction, name);
+					break;
+				case "votes":
+					votesPoll(interaction, name);
+					break;
+			}
+		} else if (subCmdGroup === "list") {
+			if (subCmd === "jams") {
+				listJams(interaction);
+			} else if (subCmd === "polls") {
+				listPolls(interaction);
+			}
+		}
+	}
+
+	register(): SlashCommandSubcommandsOnlyBuilder {
+		return new SlashCommandBuilder()
+			.setName("coding-jams")
```

Comment:
rename to just jams cause not everyone will use this bot for coding related stuff

Reply:
nah... for now this is coding related... all the text are coding related. 
maybe in the future we can generalize it